CSS Module page: CSS Shapes (#33809)
mdn/content: files/en-us/web/css/css_shapes/index.md

title: CSS shapes
slug: Web/CSS/CSS_shapes
page-type: css-module
spec-urls:
  - https://drafts.csswg.org/css-shapes-1/
  - https://drafts.csswg.org/css-shapes-2/
---

{{CSSRef}}

> The **CSS shapes** module describes geometric shapes that can be used in CSS and defines properties that make use of shapes to control the geometry of an element's float area, to apply to exclusions, and to apply to an element's content area.

Reviewer: ```suggestion
The **CSS shapes** module describes geometric shapes. It also defines CSS properties that can use the shapes to control the geometry of an element's float area; this area can then be applied to exclusions, or specify an element's content area.
```

Reviewer: As it was, this sentence was a bit confusing. I'm not 100% if this suggestion makes sense, but this is the direction I thought it should go in.

> The specification defines several ways to define shapes. Shapes enable wrapping content around a shape rather than following the rectangle of the element's box. Shapes can also be used to wrap content within the element.

Reviewer: ```suggestion
The specification defines several ways to create shapes. Content can be wrapped around or within a shape rather than following the default rectangle shape of the element's box.
```

- {{cssxref("shape-margin")}}
- {{cssxref("shape-outside")}}

> **Note:** The CSS shapes module introduces the `shape-inside` and `shape-padding` properties that have not yet been implemented.

### Data types

- {{cssxref("&lt;basic-shape&gt;")}}

### Functions

- {{cssxref("basic-shape/circle","circle()")}}
- {{cssxref("basic-shape/ellipse","ellipse()")}}
- {{cssxref("basic-shape/inset","inset()")}}
- {{cssxref("basic-shape/path","path()")}}
- {{cssxref("basic-shape/polygon","polygon()")}}
- {{cssxref("basic-shape/rect","rect()")}}
- {{cssxref("basic-shape/shape","shape()")}}
- {{cssxref("basic-shape/xywh","xywh()")}}

### Terms

- [Reference box](/en-US/docs/Web/CSS/CSS_shapes/Basic_shapes#the_reference_box)

## Guides

- [Overview of shapes](/en-US/docs/Web/CSS/CSS_shapes/Overview_of_shapes)

  - : Defining basic shapes with the `shape-margin` and `clip-path` properties, and debugging basic shapes with

>     Developer Tools

Reviewer: ```suggestion
    Developer Tools.
```

- [Shapes from box values](/en-US/docs/Web/CSS/CSS_shapes/From_box_values)

  - : Using `border-radius` curvatures and CSS box model values to create shapes.

- [Basic shapes with `shape-outside`](/en-US/docs/Web/CSS/CSS_shapes/Basic_shapes)

>   - : Creating rectangles, circles, ellipses and polygons with CSS shapes, the reference box, and the `shape-outside` property.

Reviewer: ```suggestion
  - : Creating rectangles, circles, ellipses, and polygons with CSS shapes, the reference box, and the `shape-outside` property.
```

- [Shapes from images](/en-US/docs/Web/CSS/CSS_shapes/Shapes_from_images)

>   - : Create shapes from images file with an alpha channel and CSS Gradients.

Reviewer: ```suggestion
  - : Creating shapes from semi-transparent image files and CSS Gradients.
```
